examples/flutter_gallery/lib/demo/cupertino/cupertino_picker_demo.dart

```diff
@@ -113,16 +114,17 @@ class _CupertinoPickerDemoState extends State<CupertinoPickerDemo> {
             backgroundColor: CupertinoColors.white,
             onSelectedItemChanged: (int index) {
               setState(() {
-                _selectedMinute = index;
+                _selectedMinute = index % 60;
```

Comment:
This is a little inconvenient to bubble up all the way to the user's API. Can we create a SliverChildLoopingBuilderDelegate which needs a childCount. Then when it gets asked for indices outside the range (and no negative numbers either), it modulos internally and only asks the user for a simple index?

On the surface, we can still keep the .builder constructor and let it create a looping or a non looping delegate based on a parameter. The parameter will also drive the index looping on the onSelectedItemChanged callback too.

Reply:
I don't think looping is enough to build the DatePicker later.
In the date column of the DatePicker, the child list is a continuous infinite non-looping list, and the builder has to be able to build children for that.

About exposing this to the user's API - later I will wrap all of this in the DatePicker and show the DatePicker demo in the Gallery. This current alarm demo is only temporary until DatePicker is implemented.

Reply:
Ya, I'm not saying you don't need a delegate that allows negative indices. What I meant is we can have 3 modes:

Use the builder constructor with a `childCount` and `looping = false`, you get a standard SliverChildBuilderDelegate. Use this for things like blood type where the first item isn't related to the last item. 

Use the builder constructor with no `childCount`, it asserts that `looping` can't also be true. You get a negative index delegate. Use this for things like the year where we can extend infinitely long in both directions.

Use the builder constructor with a `childCount` and `looping = true`, you get a SliverChildLoopingBuilderDelegate. The IndexedWidgetBuilder will only be asked about 0..childCount-1 and it internally deals with expanding the number to interface with the render object. Use this for things like months. 

It probably makes sense to allow `looping` for the other constructor too.

Reply:
Ideally we wouldn't expose `looping`, you'd just provide the right kind of delegate.

Reply:
(or call the right kind of constructor)

Reply:
At some point, we still have to let user choose whether to 'loop' or not, so the question is only about where to do so.
I have changed so that now when constructing a CupertinoPicker from a list, users can set a bool value called 'looping' to decide if they want to loop the list. I have also rewritten the delegate classes.
Please let me know what do you think about the new code!